smpmgr/plugins.py

```diff
@@ -0,0 +1,57 @@
+"""Runtime discovery and execution of user-provided plugins."""
+
+import logging
+import os
+from importlib.util import module_from_spec, spec_from_file_location
+from pathlib import Path
+from typing import Final, NamedTuple
+
+import typer
+
+logger: Final = logging.getLogger(__name__)
+
+
+class Plugin(NamedTuple):
+    """An SMP group plugin."""
+
+    app: typer.Typer
+    path: Path
+
+
+def get_plugins(argv: list[str]) -> tuple[Plugin, ...]:
+    """Returns a tuple of plugin paths and removes them from `argv`."""
+
+    logger.debug(f"{argv=}")
+    args: Final = argv[:]
+    paths: Final[list[Path]] = []
+    for arg in args:
+        if arg.startswith("--plugin-path="):
+            logger.debug(f"Found plugin path argument: {arg}")
+            path_str = arg.split("=")[1]
+            path = Path(path_str) if Path(path_str).is_absolute() else os.getcwd() / Path(path_str)
```

Comment:
Right here, I think it needs resolve() before the absolute check to deal with use of ~ in the path.

Reply:
Not quite, looks like it's this: https://docs.python.org/3/library/pathlib.html#pathlib.Path.expanduser